### ISSUE TITLE
Undo button should only undo last change to geometry

### ISSUE THREAD

**Issue by matterhorn103** [OpenChemistry/avogadrolibs#1207]
**Avogadro version: (please complete the following information from the About box):**
- Flatpak of version 1.97.0 
- Avogadrolibs: 1.97.0
 - Qt: 5.15.7

**Desktop version: (please complete the following information):**
 - OS: OpenSUSE Tumbleweed
 - Version 20221201

**Describe the bug**
After making a series of changes to a molecule using the Manipulation Tool, each one consisting of clicking an atom, dragging it, and then releasing the mouse button, clicking the undo button results in the whole series of changes being reversed.

Doing some manipulation, then adding a new atom, then doing more manipulations, results in two sets of changes that are undone in separate stages. However, simply changing the current tool to something else and then back to Manipulation in the middle of the series doesn't result in the series being broken up into separate undo stages, all are still undone at the same time.

This is frequently frustrating behaviour as I often only want to undo the last small change I made. Each atom added using the Draw Tool is undone individually, so in my opinion it would make sense if the Manipulation Tool matched that.

**Comment by ghutchis** [comment]
I think what you're saying is that if you manipulate, release the drag, pause, then manipulate again, you want two separate undo states. If so, that makes sense, although I'll have to think about how to do that. Technically, every step along the drag is an event, so it merges them together. In principal, you want to put a "stop merging" step in?

**Comment by matterhorn103** [comment]
Yes, that's right I think. And not just when the same atom is manipulated multiple separate times, currently if I move one atom, and then another, and then press ctrl+Z, both atoms will be returned to their original positions when they should be treated as independent changes.
